api/python/t4/api.py

```diff
@@ -395,137 +395,7 @@ def sorter(pkg_info):
-        text: indexed text of object
-        source: source document for object (what is actually stored in ElasticSeach)
-        time: timestamp for operation
-    """
-    es = _create_es()
-
-    payload = {'query': {'query_string': {
-        'default_field': 'content',
-        'query': query,
-        'quote_analyzer': 'keyword',
-        }}}
-
-    r = es.search(index=es_index, body=payload)
-
-    try:
-        results = []
-        for result in r['hits']['hits']:
-            key = result['_source']['key']
-            vid = result['_source']['version_id']
-            op = result['_source']['type']
-            meta = json.dumps(result['_source']['user_meta'])
-            size = str(result['_source']['size'])
-            text = result['_source']['text']
-
-            time = str(result['_source']['updated'])
-            results.append({
-                'key': key,
-                'version_id': vid,
-                'operation': op,
-                'meta': meta,
-                'size': size,
-                'text': text,
-                'source': result['_source'],
-                'time': time
-            })
-        results = list(sorted(results, key=lambda x: x['time'], reverse=True))
-        return results
-    except KeyError as e:
-        return r
-
-def _print_table(table, padding=2):
-    cols_width = [max(len(word) for word in col) for col in zip(*table)]
-    for row in table:
-        print((" " * padding).join(word.ljust(width) for word, width in zip(row, cols_width)))
-
-def log(key, pprint=False):
-    es = _create_es()
-
-    payload = {'query': {'constant_score': {'filter': {'term': {
-        'key': key
-    }}}}}
-
-    r = es.search(index=es_index, body=payload)
-
-    try:
-        table = []
-        for result in r['hits']['hits']:
-            if result['_source']['key'] != key:
-                continue
-            vid = result['_source']['version_id']
-            op = result['_source']['type']
-            meta = json.dumps(result['_source']['user_meta'])
-            size = str(result['_source']['size'])
-            text = result['_source'].get('text', '')
-
-            time = str(result['_source']['updated'])
-            table.append({
-                'version_id': vid,
-                'operation': op,
-                'meta': meta,
-                'size': size,
-                'text': text,
-                'time': time
-            })
-        table = list(sorted(table, key=lambda x: x['time'], reverse=True))
-        if pprint:
-            ptable = [('Date', 'Version ID', 'Operation', 'Size', 'Meta')]
-            for t in table:
-                ptable.append((t['time'], t['version_id'], t['operation'],
-                               t['size'], t['meta']))
-            _print_table(ptable)
-        return table
-    except KeyError as e:
-        return r
-
-def config(*autoconfig_url, **config_values):
+def config(*catalog_url, **config_values):
```

Comment:
why is `catalog_url` a splat here and not just an optional keyword?

Reply:
It was perhaps namespace paranoia on my part.

All args go into **catalog_url, and all kwargs go into **config_values.  Both are optional.  Otherwise, if the user called `config(catalog_url='blah')`, that would trigger the config update from the catalog.  Now it sets the value `catalog_url` in the config.

I would have preferred to have the interface be an object, but that was considered too heavyweight at the time.  The goal at the time was also to set the config via the catalog, not to set individual keys/values, but I thought k/v setting for the config was important.

```python
# As is
t4.config(<url)  # autoconfigure from catalog
t4.config(catalog_url=<url>)  # set the key `catalog_url` to the value of <url>

# otherwise
t4.config(<url)  # autoconfigure from catalog
t4.config(catalog_url=<url>)  # autoconfigure from catalog
```